packages/changeset-validations/package.json

```diff
@@ -101,7 +101,8 @@
       "types": "./dist/*.d.ts",
       "default": "./dist/*.js"
     },
-    "./addon-main.js": "./addon-main.cjs"
+    "./addon-main.js": "./addon-main.cjs",
+    "./package.json": "./package.json"
```

Comment:
This is a workaround for an unfortunate issue...

`sync-pnpm` needs to find the hard linked package resolvable from the test-app, which is like a copy of our original addon in the monorepo, just without `/dist` (which is the issue we need to fix here). But `require.resolve('ember-headless-form')` does not work, because its `main` field points to its `./dist/index.js`, which des not exist before we have actually synced. So `resolve` throws. 

Then I tried with this package that I found on npm, which basically tries to solve that issue: [resolve-pkg](https://github.com/sindresorhus/resolve-pkg). But running into https://github.com/sindresorhus/resolve-pkg/issues/9.

So the workaround here is to make `mypackage/package.json` resolvable. Not ideal, but works for now. Not sure if we can do this in any other way. We want to delegate the node resolution algorithm to node itself and not re-invent this, but there seems to be no way to say "give me the module path that you _would_ try to require, without actually throwing when it is not available".

Reply:
does this one support exports? https://www.npmjs.com/package/resolve-package-path -- it's what auto-import uses

Reply:
It seems to replicate the node module resolution, instead of delegating to node (like `require.resolve`), so _probably_ that would work! Thanks for the suggestion, will try this out!

Related: https://github.com/sindresorhus/resolve-pkg/issues/9#issuecomment-1433250154

Reply:
> will try this out!
 
It works! :tada: